content/automate/ManageIQ/Infrastructure/VM/Transform/StateMachines/VmImport.class/__methods__/submit_vm_import.rb

```diff
@@ -0,0 +1,40 @@
+#
+# Description: This method initiates VM import to given infra provider
+#
+
+module ManageIQ
+  module Automate
+    module Infrastructure
+      module VM
+        module Transform
+          module StateMachines
+            class SubmitVmImport
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                provider = @handle.vmdb(:ext_management_system, @handle.root['provider_id'])
+                @handle.log(:info, 'Submitting Import')
```

Comment:
@matobet - Could you add basic validation that the required arguments exist before calling submit_import_vm?

Here's a sample basic validation method:
def validate_root_args(arg_names)
  arg_names.each do |name|
    unless $evm.root[name].present?
      msg = "Error, required root attribute: #{name} not found"
      $evm.log(:error, msg)
      raise msg
    end
  end
end

and you can call it prior to calling execute....

validate_root_args(%w(vm name cluster_id storage_id sparse))